### DISCUSSION TITLE
Refitting On_ACID results

### DISCUSSION THREAD

**Discussion by StoyoKaramihalev** [flatironinstitute/CaImAn#1196] [Q&A] [answered]
Morning! A huge thank you for this wonderful tool, it's been really fun to explore. And thank you for bringing into my life the results of a google search for "caiman on acid". 

I've been testing out On_ACID with some 1P - new neurons pop up as the algorithm progresses over time, which is fine. But then can I refit the data with the final set of components? The ones that "appear" later in the recording (image below) are almost certainly already there in the beginning, so would be great to see what they were up to earlier. 

Incidentally, I realise the offline approach might be better suited here, but the data and RAM overheads are pretty difficult to manage. Hundreds of GB of mmap files for short (and spatially downsampled) recordings mean I can only do this on a cluster with a lot of spare room. 

Any help will be appreciated. Thanks! 

<img width="965" alt="image" src="https://github.com/flatironinstitute/CaImAn/assets/50459099/6c8acd94-f5f2-4511-a31e-997513a7278a">

**Comment by kushalkolar** [comment, marked as answer]
You should be able to pass in the spatial footprints you find a the end of OnACID-E to the OnACID constructor as `Ain`, see here: https://github.com/flatironinstitute/CaImAn/blob/9b0b79ca61f20ce93259b9833e1fe18e26d4e086/caiman/source_extraction/cnmf/online_cnmf.py#L99-L101

I wonder what you would get if you gave it the existing model instead? I would try:

```python3
oncacid_obj # from previous run

onacid_iter2 = OnACID(params=same_params, Ain=onacid_obj.estimates.A)

# not entirely sure what this would do, Johanness would have a better answer
onacid_iter2 = OnACID(params=same_params, estimates=onacid_obj.estimates)
```

**Reply by StoyoKaramihalev** [reply]
Thanks a lot! Seems to have worked!

**Reply by kushalkolar** [reply]
It would be useful if you could let us know exactly what worked and what the components look like.

**Reply by StoyoKaramihalev** [reply]
The first option: `onacid_iter2 = OnACID(params=same_params, Ain=onacid_obj.estimates.A)`. Followed by `onacid_iter2.fit_online()` - reran everything and returned an object with exactly the same components. I don't really know what goes on under the hood there, so can't say if it might've come back with some new components under other circumstances.
